dart/dynamics/BallJoint.h

```diff
@@ -54,6 +54,18 @@ class BallJoint : public MultiDofJoint<3>
   /// Destructor
   virtual ~BallJoint();
 
+  /// Convert a rotation into a 3D vector that can be used to set the positions
+  /// of a BallJoint. The positions returned by this function will result in a
+  /// relative transform of
+  /// getTransformFromParentBodyNode() * _rotation * getTransformFromChildBodyNode().inverse()
+  /// between the parent BodyNode and the child BodyNode frames when applied to
+  /// a BallJoint.
+  template <typename RotationType>
+  static Eigen::Vector3d convertToPositions(const RotationType& _rotation)
```

Comment:
It's templated because Eigen uses a variety of ways to express rotations, and using a templated argument ensures that all of them can be covered by one function. I modelled it after the Isometry3d::rotate(~) function.

It would probably still work if it used a Matrix3d argument, since Eigen is pretty smart about converting its types, but the template usage here doesn't really hurt anything, because you don't need to use the template explicitly. If you look at [an example of usage](https://github.com/dartsim/dart/blob/a3cca9d907ff2fe8e482e2864f011f9b1ee3837e/unittests/testJoints.cpp#L576) you'll see that C++ can infer the template argument so you don't even need to treat it like a templated function.